stl/inc/__msvc_int128.hpp

```diff
@@ -981,6 +981,21 @@ struct _Unsigned128 : _Base128 {
         _Word[1] |= _That._Word[1];
         return *this;
     }
+
+    template <class _Ty, enable_if_t<is_floating_point_v<_Ty>, int> = 0>
+    _NODISCARD constexpr explicit operator _Ty() const noexcept {
+        return static_cast<_Ty>(_Word[1]) * static_cast<_Ty>(18446744073709551616.0) + static_cast<_Ty>(_Word[0]);
+    }
+
+    template <class _Ty, enable_if_t<is_floating_point_v<_Ty>, int> = 0>
+    constexpr explicit _Unsigned128(const _Ty _Val) noexcept {
+        _Word[0] = static_cast<uint64_t>(_Val);
+        _Word[1] = static_cast<uint64_t>(_Val / static_cast<_Ty>(18446744073709551616.0));
+    }
+
+    _NODISCARD constexpr explicit operator size_t() const noexcept {
+        return static_cast<size_t>(_Word[0]);
+    }

```

Comment:
I don't think this is necessary, see [here](https://godbolt.org/z/Yarj8crjM). I believe the explicit-convertibility is already covered by the test file.

Reply:
If you removed this, then the following code will not compile in the latest vs preview ver:
```c++
#include <__msvc_int128.hpp>
void test() {
	using std::_Unsigned128;
	double a=0;
	size_t a_s=(size_t)(_Unsigned128)(a);
}
```
I'm not sure what the problem is, but as far as I can tell it can only be solved by appending this overload.

Reply:
> If you removed this, then the following code will not compile in the latest vs(preview ver) :
> 
> ```c++
> #include <__msvc_int128.hpp>
> void test() {
> 	double a=0;
> 	size_t a_s=(size_t)(std::_Signed128)(a);
> }
> ```
> 
> I'm not sure what the problem is, but as far as I can tell it can only be solved by appending this overload.

I've tried this example in 17.6 preview 1 and it compiles with MSVC & Clang.

Reply:
> https://github.com/microsoft/STL/pull/3559#discussion_r1133079325

can you try `debug|x64`?

Reply:
[ConsoleApplication1.zip](https://github.com/microsoft/STL/files/10948558/ConsoleApplication1.zip)
try this?

Reply:
> [ConsoleApplication1.zip](https://github.com/microsoft/STL/files/10948558/ConsoleApplication1.zip) try this?

Strangely, this compiled with the `__msvc_int128.hpp` modified in https://github.com/steve02081504/STL/commit/a711aa9fde1b4e64b44c6eb6c54dd204f4c9fa49 during my local test.

Reply:
(reinstalling vs)

Reply:
Reinstalling vs didn't work for me😥, and the version without `operator size_t` wouldn't pass the test code in my environment.
Considering the code has no substantial impact, I'm inclined to keep it to facilitate people like me.🙏
Is the compiler you are using pure msvc?

Reply:
noted at [`54d1e2a` (#3559)](https://github.com/microsoft/STL/pull/3559/commits/54d1e2a2d8e3873c15624022f5f532c3ad19ab8c)
mark as resolved.